homeassistant/components/sharkiq/vacuum.py

```diff
@@ -198,6 +217,35 @@ async def async_locate(self, **kwargs: Any) -> None:
         """Cause the device to generate a loud chirp."""
         await self.sharkiq.async_find_device()
 
+    async def async_clean_room(self, rooms: list[str], **kwargs: Any) -> None:
+        """Clean a specific room."""
+        if len(rooms) == 0:
+            raise exceptions.HomeAssistantError("No rooms to clean were provided.")
+        rooms_to_clean = []
+        valid_rooms = []
+        if self.available_rooms is not None:
+            valid_rooms = self.available_rooms
+        all_rooms_reachable = True
+        for room in rooms:
+            if room in valid_rooms:
+                rooms_to_clean.append(room)
+            elif room.capitalize() in valid_rooms:
+                rooms_to_clean.append(room.capitalize())
+            else:
+                all_rooms_reachable = False
+                LOGGER.error("Room not reachable: %s", room)
+
+        if all_rooms_reachable:
+            LOGGER.info("Cleaning room(s): %s", rooms_to_clean)
+            await self.sharkiq.async_clean_rooms(rooms_to_clean)
+        else:
+            LOGGER.error("Invalid room selection - service not run")
+            raise exceptions.HomeAssistantError(
+                "One or more of the rooms listed is not available to your vacuum. "
+                "Make sure all rooms match the Shark App including capitalization."
+            )
```

Comment:
```suggestion
        LOGGER.info("Cleaning room(s): %s", rooms_to_clean)
        await self.sharkiq.async_clean_rooms(rooms_to_clean)
```

Reply:
I think this is cleaned up to your recommendation now